Adapting installation process to macOS AND Linux

DanielOberlechner opened the pull request:
Fixes #5306 

Firstly there was always talk about macOS but not Linux. This is probably for some people not so easy to understand that it is quite the same with some exceptions to it.

I rewrote some parts of this readme and modified suited for Linux and macOS.

(should be good also for SEO)

Kind regards,
Daniel

## Description

_What did you add/update/change?_

## Type of suggestion

* [ ] Typo/grammar fix
  [X] Updated outdated content
* [ ] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)
Umbraco V12


## Deadline (if relevant)

_When should the content be published?_

ASAP :D

sofietoft commented on 12/umbraco-cms/fundamentals/setup/install/umbraco-on-linux-macos.md:
```diff
@@ -21,10 +21,10 @@ With the templates installed on your system, it is now possible to create Umbrac
 Once you create a new project it will use SQLite by default on Mac.
 
-If you want to use an SQL server database, you will need to [set up Docker](https://creativewebspecialist.co.uk/2021/09/07/how-to-run-netcore-umbraco-cms-on-a-macbook/) to be able to use SQL Server with Mac.
+If you want to use an SQL server database, you will need to [set up Docker](https://creativewebspecialist.co.uk/2021/09/07/how-to-run-netcore-umbraco-cms-on-a-macbook/) to be able to use SQL Server on Linux or macOS.
```
"... to use SQL Server ..." was mentioned twice. Removed one of them! What do you think? Does it still make sense?

```suggestion
If you want to use an SQL server database, you will need to [set up Docker](https://creativewebspecialist.co.uk/2021/09/07/how-to-run-netcore-umbraco-cms-on-a-macbook/).
```

sofietoft commented on 12/umbraco-cms/fundamentals/setup/install/umbraco-on-linux-macos.md:
```diff
@@ -21,10 +21,10 @@ With the templates installed on your system, it is now possible to create Umbrac
 
 To create new projects using Visual Studio, you can use the [Install using Visual Studio](visual-studio.md) guide.
 
 Once you create a new project it will use SQLite by default on Mac.
```
Do you happen to know if SQLite is the default for Linux as well? 😅 
Otherwise, I'll run it by our developers. Let em know 😁

sofietoft reviewed (commented):
Great PR @DanielOberlechner 💪 

I've added a couple of comments - let me know what you think!

DanielOberlechner reviewed (commented):
👍

sofietoft commented:
Thanks for considering my suggestion @DanielOberlechner ! 💪 

Did you see my comment about the SQLite part, and whether that's also the default for Linux?

DanielOberlechner commented:
https://docs.umbraco.com/umbraco-cloud/set-up/working-with-mac

Here it seems to be the same case with the documentation like above.
I'll try shortly to release an update. :)

sofietoft commented:
Thanks for adding this as well.
I'll make sure we take a look at it later this week.

Regarding the warnings - looks like it's related to a long sentence here: https://github.com/umbraco/UmbracoDocs/pull/5307/files#diff-c58bcadf3ab1d36340a79e54ffaec8ea8dbe68230127884f37339bd93185cbedR3

We have some styleguides in place, which, when not followed, will give you some warnings.
Usually you can read the warnings to learn how to resolve - otherwise, have a look here: https://docs.umbraco.com/welcome/contribute/style-guide

DanielOberlechner commented:
@sofietoft 
I understand the warning thing, but the check's wouldn't fail because of a warning, or they do?
I read in the log something others failed and therefore the check wasn't completed.

But don't take it as granted.

For example this is the first watcher's real error message:
{"message": "[UmbracoDocs.Acronyms] ' RTE' has no definition", "location": {"path": "11/umbraco-forms/editor/creating-a-form/README.md", "range": {"start": {"line": 94, "column": 105}}}, "severity": "ERROR"}

And such ones are a lot of in this run.

Kind regards,
Daniel

sofietoft commented:
Ah, I see!
The Vale Linter runs through the entire document, but will only fail if it finds issues (warnings/errors) in the file that's being changed in the PR.
So, you only need to worry about the warnings/errors that show up in the file under the "Files changed" tab in the PR 💪 

I hope it makes sense!

sofietoft commented on umbraco-cloud/set-up/working-with-linux-macos.md:
```diff
@@ -1,10 +1,8 @@
+# Working with Linux/macOS
 
-One of the features built into Umbraco Cloud is the ability to work locally with your Umbraco site - without having a Windows machine with a local web server installed on it. This enables people on Mac or Linux-based OS to use their favorite editor to modify code in their Umbraco Cloud site.
+One of the features built into Umbraco Cloud is the ability to work locally with your Umbraco site - without having a Windows machine with a local web server installed on it. This enables people on macOS or Linux-based OS to use their favorite editor to modify code in their Umbraco Cloud site.
```
```suggestion
One of the features built into Umbraco Cloud is the ability to work locally with your Umbraco site. This can be done without having a Windows machine with a local web server installed on it. This enables people on macOS or Linux-based OS to use their favorite editor to modify code in their Umbraco Cloud site.
```

sofietoft reviewed (changes requested):
So, I think I identified the reason why the checks here are still failing.
There was one long sentence remaining in the `working-with-linux-macos.md` article, in line 3.
I've added a suggestion to shorten it.

When checking where the checks have failed, you need to look only at the "Files changed" tab. No need to worry about all the many warnings you see in the logs, as the linter will always go through all files in the repo.
I hope it makes sense!

Also, I noticed that you didn't rename the `running-umbraco-on-a-mac.md` files for versions 10 and 11, like you did for 12.
Could you take a look at that as well? 💪

DanielOberlechner commented:
@sofietoft 
Ah sorry, my fault.
I fixed by now the remaining files and renamed them accordingly. 

By now everything should be up2date and working.

Do I have anything left to do,
or do I wait until someone merges this branch into master?

Kind regards,
Daniel

DanielOberlechner reviewed (commented):
is ok

sofietoft commented on umbraco-cloud/set-up/working-locally.md:
```diff
@@ -4,7 +4,7 @@ description: "In this article we explain how you can work with a local clone of
 # Working with a Local Clone
 
-To work with a local copy of your site, you'll need to use Windows with a local web server installed (like WebMatrix/IIS). If you're using Mac, see the [Working with Mac](working-with-mac.md) article.
+To work with a local copy of your site, you'll need to use Windows with a local web server installed (like WebMatrix/IIS). If you're using macOS or Linux based OS, see the [Working with Linux/macOS](working-with-linux-macos.md) article.
```
```suggestion
To work with a local copy of your site, you'll need to use Windows with a local web server installed (like WebMatrix/IIS). If you're using macOS or Linux-based operating system, see the [Working with Linux/macOS](working-with-linux-macos.md) article.
```

sofietoft reviewed (commented):
The changes looks great @DanielOberlechner ! 💪 
Just one more suggestion from my side.

I'll also need to add rewrites for the files being renamed here. 
I'll see whether I can add those directly to this PR here, or if I'll have to do it once it's been merged.

sofietoft commented:
Hi @DanielOberlechner ! 👋 

Did you have a chance to take a look at the redirects part for this PR - see my previous comment.

I'd be more than happy to do it - just let me know!

DanielOberlechner commented:
> It doesn't look like I have permission to push to your branch.
> 
> Would you be up for adding a few more lines to the PR? I can also add these after we've merged this one - let me know!
> 
> `set-up/working-with-mac: set-up/working-with-linux-macos.md`
> 
> Needs to be added to the `/umbraco-cloud/.gitbook.yaml` file under` redirects`.
> 
> `fundamentals/setup/install/running-umbraco-on-a-mac: fundamentals/setup/install/running-umbraco-on-linux-macos.md` Needs to be added in 3 different places:
> 
> * `/10/umbraco-cms/.gitbook.yaml` under `redirects`
> * `/11/umbraco-cms/.gitbook.yaml` under `redirects`
> * `/12/umbraco-cms/.gitbook.yaml` under `redirects`

@sofietoft 
Hey Sofie, I'm sorry your comment from 5 day's ago I did not see! It's my first real pull request and I'm not 100% familiar with the procedere.

I created now the changes you proposed.

Now everything is done. :)

sofietoft commented:
No worries 😄 

Looks great! Thanks for putting in all the effort to get this one done 💪 
It all looks great.
I'll take a final look at the previews on the docs platform, and then I'll get it merged!

sofietoft commented:
Ah, yes! I think I wasn't clear enough in my last comment.

The file that you've renamed in version 12 of the docs - it's missing the "running-" part.
The file should be called `running-umbraco-on-linux-macos.md`, like the other rights?

Currently it's called `umbraco-on-linux-macos.md`

I hope it makes sense 🤞

DanielOberlechner commented:
@sofietoft ah ok, got it. file is renamed :)

sofietoft commented:
Excellent @DanielOberlechner ! 💪 

It all checks out now on the docs platform.
I'll get it merged.

Thanks again for putting in all the effort to get this PR written and sorted!
Really appreciate it 😄 
Hope to see you back here with even more contributions in the near future!
